### ISSUE TITLE
Whoops Session Time Out On Place Order

### ISSUE THREAD

**Issue by dbltoe** [zencart/zencart#3565]
Less hair now
Complete beta with demos and no mods allows Place Order for the one customer.
Complete beta no demos, no mods, imported database with responsive_classic gives the Whoops session time out even for newest of customers.  Had switched to responsive_classic to avoid mods.
Not sure if OPC related but unable to totally remove it from store.
Same results with OPC disabled in any setting.
Tried several combos in process of elimination to no avail.
Bottom line, the only one that seems to work is beta, demos, and no mods -- plain Jane
Thoughts?

**Comment by lat9** [comment]
I'm not seeing this on the current repository beta and OPC v2.3.0, 2.3.1 or 2.3.2.

**Comment by dbltoe** [comment]
Yep, thought we worked out any kinks in 157 OPC but wanted to mention it was there and set to false.
I’ve checked database structure to see if I missed an update but all checked out

**Comment by dbltoe** [comment]
Been fighting this for a couple of days.  No logs, no console error. 👎

**Comment by drbyte** [comment]
> Complete beta no demos, no mods, imported database

"imported database"...

Do you mean "imported everything including all configs"? (ie: things that could possibly interfere because they've changed the defaults that appear to work when just "plain jane"?

**Comment by drbyte** [comment]
Could it be related to the EMP/login-as-customer changes?

**Comment by drbyte** [comment]
The "whoops timeout" is typically caused by session expiry. It can also occur when something "triggers" session-expiry such as credit-card-slamming or trying to use outdated session cookie properties. 
The login-as-customer does interact with sessions; perhaps prior configs from the old mod might be interfering?
Switching domain names between "www." and non-"www." prefixes will invalidate session cookies. Is there a mod or a custom link that's using different urls?

**Comment by dbltoe** [comment]
Can't find any strays in the Only thing I can find is the layout of the configuration table.
Store has Type mediumtext while beta has text
Store has configuration_tab that is not in beta filled with General

> Could it be related to the EMP/login-as-customer changes?

Had verified removal of 1.5.# EMP

**Comment by dbltoe** [comment]
Is it possible that I missed this?`
INSERT IGNORE INTO configuration (configuration_title, configuration_key, configuration_value, configuration_description, configuration_group_id, sort_order, date_added) VALUES ('Customer <em>Place Order</em>: Admin Profiles', 'EMP_LOGIN_ADMIN_PROFILE_ID', '1', 'Identify the admin <em>User Profile IDs</em> that are permitted to use the <em>Place Order</em> feature on the customers\' listing &mdash; all admins that are in these profiles are permitted. Enter the value as a comma-separated list (intervening blanks are OK) of Admin Profile IDs, e.g. <b>1, 2, 3</b>. Set the value to 0 to disable the <em>Admin Profiles</em> feature.<br /><br /><b>Default: 1 (All Superusers)</b><br />', 1, 301, now());
UPDATE configuration SET configuration_title = 'Customer <em>Place Order</em>: Admin Profiles', configuration_description = 'Identify the admin <em>User Profile IDs</em> that are permitted to use the <em>Place Order</em> feature on the customers\' listing &mdash; all admins that are in these profiles are permitted. Enter the value as a comma-separated list (intervening blanks are OK) of Admin Profile IDs, e.g. <b>1, 2, 3</b>. Set the value to 0 to disable the <em>Admin Profiles</em> feature.<br /><br /><b>Default: 1 (All Superusers)</b><br />' WHERE configuration_key = 'EMP_LOGIN_ADMIN_PROFILE_ID' LIMIT 1;

INSERT IGNORE INTO configuration (configuration_title, configuration_key, configuration_value, configuration_description, configuration_group_id, sort_order, date_added, set_function) VALUES ('Customer <em>Place Order</em>: Passwordless Login', 'EMP_LOGIN_AUTOMATIC', 'true', 'Login directly to store without entering credentials', 1, 302, now(), 'zen_cfg_select_option(array(\'true\', \'false\'),');
UPDATE configuration SET configuration_title = 'Customer <em>Place Order</em>: Passwordless Login', configuration_description = 'Login directly to store without entering credentials' WHERE configuration_key = 'EMP_LOGIN_AUTOMATIC' LIMIT 1;
`
Or maybe a miss on the prior upgrade to 1.5.6?

**Comment by Zen4All** [comment]
> Store has  that is not in beta filled with General
The configuration_tab is not default Zen Cart. It comes with numinix add-ons.

**Comment by dbltoe** [comment]
THANKS gives me a new direction for a cure.

**Comment by dbltoe** [comment]
I've gone through the configuration table item by item.  Found a couple of strays that had no effect.  Did find several entries that were in the beta site's configuration table but not in the store's.
Any suggestions on how to "catch up" without endangering data other than just back up and "let 'er rip tater chip!"?

**Comment by dbltoe** [comment]
Wondering if some of the Display Log mentions could be playing with my situation.
I went through a beta database and the site's database with Navicat and brought the beta up to synchronization with the site.  Found one or two hangers on in the process.
Still no joy and placing error reporting in both configs gives no error other than the Whoops.
👎

**Comment by drbyte** [comment]
The only way I've been able to trigger "Whoops, session timeout when trying to Order" is with a precarious combination of factors:
- browser in incognito (aka private-browsing) mode: ie: where cookies aren't set
- .htaccess redirects in place for "human friendly URLs"
- v1.5.5
- clicking Add To Cart while not logged in, and in incognito mode, triggers the Timeout screen

**Comment by dbltoe** [comment]
The only possibility out of those items is that this cart was started at 1.5.0.
It always seems the freebies are the hardest.  Non-profit and, I guess, several "experts" have had their hand in.  Expert I am NOT.

**Comment by drbyte** [comment]
Is it browser-specific? device-specific?

**Comment by dbltoe** [comment]
Looks as if it has to do with the loading of the login page and OPC may be part of the problem.

I am getting:

> Uncaught TypeError:  Cannot read property 'email_address' of undefined at onload (index.php?main_page=login:735
The page source for 735 shows

> <body id="loginBody" onload="update_zone(document.createAccountForm); document.loginForm.email_address.focus();">

In a bare bones beta, the line shows exactly the same.

An inspect of login shows opc_label is being used.

<body id="loginBody" onload="update_zone(document.createAccountForm); document.loginForm.email_address.focus();">  shows in the console with a down arrow under the end of the line and an X in a red circle

**Comment by dbltoe** [comment]
With probably the fifth beta setup, I've finally determined OPC 2.3.2 ( prior to 1.5.7 changes) to be causing a login problem ONLY when Guest Checkout is set to true.

When not set to true, BOTH Place Order and the EMP are working.

Closing this to work with lat9 on a fix for 1.5.7.
